core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -166,8 +166,12 @@ public abstract class SqlImplementor {
 
   protected final Map<CorrelationId, Context> correlTableMap = new HashMap<>();
 
-  /** Private RexBuilder for short-lived expressions. It has its own
-   * dedicated type factory, so don't trust the types to be canonized. */
+  /**
+   * Private RexBuilder for short-lived expressions. It has its own
+   * dedicated type factory, so don't trust the types to be canonized.
+   *
+   * @deprecated Replaced by {@link RexBuilder#DEFAULT}. */
+  @Deprecated // to be removed before 2.0
```

Comment:
I am not sure about this change, though

Reply:
This change is necessary. I tried to obtain `RexBuilder` through `implementer().RexBuilder`, but SimpleContext may throw UnsupportedOperaException in some cases.

Reply:
Well, I approved the PR.
But it would be good to know why this method needs to be essentially deprecated (because it can't be used).

Reply:
I am concerned that other projects have implemented this interface and obtained this RexBuilder instance through `implementer(). RexBuilder`.We don't need it anymore. To maintain compatibility, I added @ Deprecated.